### DISCUSSION TITLE
Support for recently added CSS functionality container-queries

### DISCUSSION THREAD

**Discussion by treponat** [less/less.js#3759] [Ideas]
Recently some of the browser started to support a new feature that would allow to create responsive design based on the parent element rather than client's viewport. 
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Container_Queries
It would be wonderful if LESS would also start to fully support this feature. 
Currently you can use container functionality in .less files, but the hierarchical structure of css blocks is not working as expected for the container syntax. 
for example
```
.widget.discoverresults, .widget.repositoriesresults {
        container-type: inline-size;
        @container (max-width: 350px) {           
                    .cite {
                        .wdr-authors {
                            display: none;
                         }
                }
         }
}
    
```

would produce this css blocks 

```
.widget.discoverresults,
.widget.repositoriesresults {
  container-type: inline-size;
}
@container (max-width: 350px) {
  .cite .wdr-authors {
    display: none;
  }
}
```
instead of 
```
.widget.discoverresults,
.widget.repositoriesresults {
	container-type: inline-size;
}

@container (max-width: 350px) {

	.widget.discoverresults,
	.widget.repositoriesresults {
		.cite .wdr-authors {
			display: none;
		}
	}
}
```

**Comment by tatof** [comment]
Yes container queries for less support would be awesome! Now i need to write the container queries at the bottom of the page with 10000 different classes :( kinda sucks.

When will support drop?

**Comment by carletonm** [comment]
Yes! I'm pulling my hair out about this issue. Hope we can get support for this soon :)

**Comment by matthew-dean** [comment]
Tracking here: https://github.com/less/less.js/issues/3766

**Reply by tatof** [reply]
Dont think this is the same question/problem.

**Reply by matthew-dean** [reply]
Somehow a digit got erased. Fixed.

**Reply by tatof** [reply]
Thanks for updating :)
